testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/cookie/DefaultCookieVersionTestCase.java

```diff
@@ -106,11 +118,52 @@ public void testDefaultCookieVersion() throws Exception {
             response = httpclient.execute(new HttpPost(cookieURL.toURI() + "CookieServlet"));
             Header[] cookies = response.getHeaders("set-cookie");
 
+            Assert.assertTrue(cookies.length > 0);
+
             for (Header i : cookies) {
                 String value = i.getValue();
-                Assert.assertTrue(value + Arrays.toString(cookies), value.toLowerCase(Locale.ENGLISH).contains("version=1"));
+                Assert.assertTrue(value + Arrays.toString(cookies), value.toLowerCase(Locale.ENGLISH).contains
+                        ("version=" + cookieVersion));
+            }
+        }
+    }
+
+    private void commonSendCookieVersion(int cookieVersion) throws IOException, URISyntaxException {
+        configureDefaultCookieVersion(cookieVersion);
+
+        BasicCookieStore basicCookieStore = new BasicCookieStore();
+        BasicClientCookie cookie = new BasicClientCookie("testCookie", "testCookieValue");
+        cookie.setVersion(cookieVersion);
+        cookie.setDomain(cookieURL.getHost());
+        basicCookieStore.addCookie(cookie);
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().setDefaultCookieStore(basicCookieStore)
+                .build()) {
+            HttpResponse response = httpclient.execute(new HttpGet(cookieURL.toURI() + "CookieEchoServlet"));
+            if (response.getEntity() != null) {
+                Assert.assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+                Assert.assertEquals(cookieVersion + "", EntityUtils.toString(response.getEntity()));
```

Comment:
This test is not valid, the default cookie version is only applicable when the browser is sending the cookie.

Reply:
I am not sure, whether I follow here. Client in this test creates a cookie by itself and sends it to the server. In servlet there is only checked what cookie version was determined by Undertow for such client cookie. What am I missing?